### PR TITLE
Task/tec 3280 dynamic labels

### PR DESCRIPTION
**Pull request by paulmskim**
[TEC-3280]

Screencast: https://drive.google.com/open?id=1ajz2KCWpOwQga-qu-vS0E1tNZP2FTWKW

[TEC-3280]: https://moderntribe.atlassian.net/browse/TEC-3280

### CODE REVIEW

File: src/Tribe/Views/V2/Messages.php [hunk 1]
```diff
@@ -120,7 +120,26 @@ public static function for_key( $key, ...$values ) {
...
+		 * Filters the array of keys of the messages that need the events label.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $need_events_label_keys Array of keys of the messages that need events label.
+		 */
+		$need_events_label_keys = apply_filters( 'tribe_events_views_v2_messages_need_events_label_keys', $need_events_label_keys );
+
+		if ( in_array( $key, $need_events_label_keys ) ) {
+			array_unshift( $values, tribe_get_event_label_plural() );
```
Comment (paulmskim) [inline src/Tribe/Views/V2/Messages.php]: not sure if this is the best way to do this, also see https://github.com/moderntribe/events-pro/pull/1512 for the counterpart to the filter.

File: src/Tribe/Views/V2/View.php [hunk 2]
```diff
@@ -1940,7 +1940,13 @@ protected function get_ical_data() {
 		 *
 		 * @param string $text The default link text, which is "Export Events".
 		 */
-		$link_text  = apply_filters( 'tribe_events_ical_export_text', __( 'Export Events', 'the-events-calendar' ) );
+		$link_text  = apply_filters(
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :warning: Equals sign not aligned correctly; expected 1 space but found 2 spaces

<sub>hash: 6c86fda4c7c160febf39a1eb0a4eeb5a</sub>

File: src/Tribe/Views/V2/View.php [hunk 3]
```diff
@@ -1940,7 +1940,13 @@ protected function get_ical_data() {
...
+			'tribe_events_ical_export_text',
+			sprintf(
+				__( 'Export %s', 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/View.php]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: d1837ae337c874ee4439f72cf47048ee</sub>

File: src/views/v2/components/events-bar/search/keyword.php [hunk 4]
```diff
@@ -29,7 +29,7 @@ class="tribe-common-form-control-text__input tribe-events-c-search__input tribe-
 		id="tribe-events-events-bar-keyword"
 		name="tribe-events-views[tribe-bar-search]"
 		value="<?php echo esc_attr( tribe_events_template_var( [ 'bar', 'keyword' ], '' ) ); ?>"
-		placeholder="<?php esc_attr_e( 'Search for events', 'the-events-calendar' ); ?>"
+		placeholder="<?php printf( esc_attr__( 'Search for %s', 'the-events-calendar' ), tribe_get_event_label_plural() ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/components/events-bar/search/keyword.php]: * :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'tribe_get_event_label_plural'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')'

<sub>hash: e68216dfb66c1b0e745c4fe9806f8730</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 5]
```diff
@@ -76,7 +79,16 @@
 	>
 		<h3 class="tribe-events-calendar-month__day-date tribe-common-h6 tribe-common-h--alt">
 			<span class="tribe-common-a11y-visual-hide">
-				<?php echo esc_html( sprintf( _n( '%s event', '%s events', $day['found_events'], 'the-events-calendar' ), number_format_i18n( $day['found_events'] ) ) ); ?>,
+				<?php
+				echo esc_html(
+					sprintf(
+						_n( '%1$s %2$s', '%1$s %3$s', $day['found_events'], 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: * :warning: Mismatched placeholders is probably an error
* :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: 27659638292f1d511a260755fcdc42e1</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 6]
```diff
@@ -76,7 +79,16 @@
...
+						number_format_i18n( $day['found_events'] ),
+						$events_label_singular,
+						$events_label_plural
+					)
+				);
+				?>,
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: 27a9947e805f15f556cf135c56570d40</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 7]
```diff
@@ -88,15 +100,15 @@ class="tribe-events-calendar-month__day-date-daynum"
 		<?php if ( ! empty( $day['featured_events'] ) ): ?>
 			<em
 				class="tribe-events-calendar-month__mobile-events-icon tribe-events-calendar-month__mobile-events-icon--featured"
-				aria-label="<?php esc_attr_e( 'Has featured events', 'the-events-calendar' ); ?>"
-				title="<?php esc_attr_e( 'Has featured events', 'the-events-calendar' ); ?>"
+				aria-label="<?php echo esc_attr( sprintf( __( 'Has featured %s', 'the-events-calendar' ), $events_label_plural ) ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: c617965ffdadb9875450f041fd9c405a</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 8]
```diff
@@ -88,15 +100,15 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				title="<?php echo esc_attr( sprintf( __( 'Has featured %s', 'the-events-calendar' ), $events_label_plural ) ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: e3d7c0ccd13024ce1d0cee0a2e1a0637</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 9]
```diff
@@ -88,15 +100,15 @@ class="tribe-events-calendar-month__day-date-daynum"
...
 			>
 			</em>
 		<?php elseif ( ! empty( $day['found_events'] ) ) : ?>
 			<em
 				class="tribe-events-calendar-month__mobile-events-icon tribe-events-calendar-month__mobile-events-icon--event"
-				aria-label="<?php esc_attr_e( 'Has events', 'the-events-calendar' ); ?>"
-				title="<?php esc_attr_e( 'Has events', 'the-events-calendar' ); ?>"
+				aria-label="<?php echo esc_attr( sprintf( __( 'Has %s', 'the-events-calendar' ), $events_label_plural ) ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: 86a08212f2b2f1a2c8fcc3209e8b8152</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 10]
```diff
@@ -88,15 +100,15 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				title="<?php echo esc_attr( sprintf( __( 'Has %s', 'the-events-calendar' ), $events_label_plural ) ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: 28558874fcf8ecda82bfa3ec0b29fb72</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 11]
```diff
@@ -108,7 +120,16 @@ class="tribe-events-calendar-month__day-cell tribe-events-calendar-month__day-ce
 	>
 		<h3 class="tribe-events-calendar-month__day-date tribe-common-h4">
 			<span class="tribe-common-a11y-visual-hide">
-				<?php echo esc_html( sprintf( _n( '%s event', '%s events', $day['found_events'], 'the-events-calendar' ), number_format_i18n( $day['found_events'] ) ) ); ?>,
+				<?php
+				echo esc_html(
+					sprintf(
+						_n( '%1$s %2$s', '%1$s %3$s', $day['found_events'], 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: * :warning: Mismatched placeholders is probably an error
* :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: a608eea9ab64d2e464255f14f9fee083</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 12]
```diff
@@ -108,7 +120,16 @@ class="tribe-events-calendar-month__day-cell tribe-events-calendar-month__day-ce
...
+						number_format_i18n( $day['found_events'] ),
+						$events_label_singular,
+						$events_label_plural
+					)
+				);
+				?>,
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: 326b9be8b46f9d7b1a6f6a2000bce91f</sub>

File: src/views/v2/components/events-bar/search/keyword.php [hunk 13]
```diff
@@ -29,7 +29,23 @@ class="tribe-common-form-control-text__input tribe-events-c-search__input tribe-
 		id="tribe-events-events-bar-keyword"
 		name="tribe-events-views[tribe-bar-search]"
 		value="<?php echo esc_attr( tribe_events_template_var( [ 'bar', 'keyword' ], '' ) ); ?>"
-		placeholder="<?php esc_attr_e( 'Search for events', 'the-events-calendar' ); ?>"
-		aria-label="<?php printf( esc_attr__( 'Enter Keyword. Search for %s by Keyword.', 'the-events-calendar' ), tribe_get_event_label_plural() ); ?>"
+		placeholder="<?php
```
Comment (tr1b0t) [inline src/views/v2/components/events-bar/search/keyword.php]: :no_entry_sign: Opening PHP tag must be on a line by itself

<sub>hash: 835367c59227983ac351c6130b10e595</sub>

File: src/views/v2/components/events-bar/search/keyword.php [hunk 14]
```diff
@@ -29,7 +29,23 @@ class="tribe-common-form-control-text__input tribe-events-c-search__input tribe-
...
+			echo esc_attr(
+				sprintf(
+					/* translators: %s: Events (plural). */
+					__( 'Search for %s', 'the-events-calendar' ),
+					tribe_get_event_label_plural()
+				)
+			);
+		?>"
```
Comment (tr1b0t) [inline src/views/v2/components/events-bar/search/keyword.php]: * :no_entry_sign: Line indented incorrectly; expected 3 tabs, found 2
* :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: 100ac72eb2a8cbfac0397a32472ad0b1</sub>

File: src/views/v2/components/events-bar/search/keyword.php [hunk 15]
```diff
@@ -29,7 +29,23 @@ class="tribe-common-form-control-text__input tribe-events-c-search__input tribe-
...
+		aria-label="<?php
```
Comment (tr1b0t) [inline src/views/v2/components/events-bar/search/keyword.php]: :no_entry_sign: Opening PHP tag must be on a line by itself

<sub>hash: a2e74586adf340974b563b88d2fe8f28</sub>

File: src/views/v2/components/events-bar/search/keyword.php [hunk 16]
```diff
@@ -29,7 +29,23 @@ class="tribe-common-form-control-text__input tribe-events-c-search__input tribe-
...
+			echo esc_attr(
+				sprintf(
+					/* translators: %s: Events (plural). */
+					__( 'Enter Keyword. Search for %s by Keyword.', 'the-events-calendar' ),
+					tribe_get_event_label_plural()
+				)
+			);
+		?>"
```
Comment (tr1b0t) [inline src/views/v2/components/events-bar/search/keyword.php]: * :no_entry_sign: Line indented incorrectly; expected 3 tabs, found 2
* :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: 401452a8fa5362f81e0f5975e2d0be1f</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 17]
```diff
@@ -76,7 +79,18 @@
 	>
 		<h3 class="tribe-events-calendar-month__day-date tribe-common-h6 tribe-common-h--alt">
 			<span class="tribe-common-a11y-visual-hide">
-				<?php echo esc_html( sprintf( _n( '%s event', '%s events', $day['found_events'], 'the-events-calendar' ), number_format_i18n( $day['found_events'] ) ) ); ?>,
+				<?php
+				echo esc_html(
+					sprintf(
+						/* translators: %1$s: number of events, %2$s: Event (singular), %3$s: Events (plural). */
+						_n( '%1$s %2$s', '%1$s %3$s', $day['found_events'], 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :warning: Mismatched placeholders is probably an error

<sub>hash: a1238f11d490a7825ead05e6b639261d</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 18]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
 		<?php if ( ! empty( $day['featured_events'] ) ): ?>
 			<em
 				class="tribe-events-calendar-month__mobile-events-icon tribe-events-calendar-month__mobile-events-icon--featured"
-				aria-label="<?php esc_attr_e( 'Has featured events', 'the-events-calendar' ); ?>"
-				title="<?php esc_attr_e( 'Has featured events', 'the-events-calendar' ); ?>"
+				aria-label="<?php
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Opening PHP tag must be on a line by itself

<sub>hash: 55103c69c99d46626ec0041311284b14</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 19]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+					echo esc_attr(
+						sprintf(
+							/* translators: %s: Events (plural). */
+							__( 'Has featured %s', 'the-events-calendar' ),
+							$events_label_plural
+						)
+					);
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Line indented incorrectly; expected at least 7 tabs, found 5

<sub>hash: b7deadf33276b2f8ffc38d613e305d97</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 20]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: * :no_entry_sign: Line indented incorrectly; expected 7 tabs, found 4
* :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: dcc24da3f8c5c60ced774d4d173e3197</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 21]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				title="<?php
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Opening PHP tag must be on a line by itself

<sub>hash: 100795e663070e7d74e74e82a36dd74f</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 22]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+					echo esc_attr(
+						sprintf(
+							/* translators: %s: Events (plural). */
+							__( 'Has featured %s', 'the-events-calendar' ),
+							$events_label_plural
+						)
+					);
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Line indented incorrectly; expected at least 5 tabs, found 5

<sub>hash: 749a0f16f0d73c58a0c96e6a078668fb</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 23]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: * :no_entry_sign: Line indented incorrectly; expected 6 tabs, found 4
* :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: e50381091ffd5b6f361b547e1e74a119</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 24]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
 			>
 			</em>
 		<?php elseif ( ! empty( $day['found_events'] ) ) : ?>
 			<em
 				class="tribe-events-calendar-month__mobile-events-icon tribe-events-calendar-month__mobile-events-icon--event"
-				aria-label="<?php esc_attr_e( 'Has events', 'the-events-calendar' ); ?>"
-				title="<?php esc_attr_e( 'Has events', 'the-events-calendar' ); ?>"
+				aria-label="<?php
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Opening PHP tag must be on a line by itself

<sub>hash: 4a762b00c4b654536b61365950b9fc67</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 25]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+					echo esc_attr(
+						sprintf(
+							/* translators: %s: Events (plural). */
+							__( 'Has %s', 'the-events-calendar' ),
+							$events_label_plural
+						)
+					);
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Line indented incorrectly; expected at least 7 tabs, found 5

<sub>hash: 0383f5a5ab10a87c3845f023cfde358c</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 26]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: * :no_entry_sign: Line indented incorrectly; expected 7 tabs, found 4
* :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: dba6e637f3766b17152b1f94c2e5f674</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 27]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				title="<?php
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Opening PHP tag must be on a line by itself

<sub>hash: 2a0209ce41cebe65557fc9b6ccdb1358</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 28]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+					echo esc_attr(
+						sprintf(
+							/* translators: %s: Events (plural). */
+							__( 'Has %s', 'the-events-calendar' ),
+							$events_label_plural
+						)
+					);
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :no_entry_sign: Line indented incorrectly; expected at least 5 tabs, found 5

<sub>hash: 1c143f76e0da0c37f2da297c256b6b95</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 29]
```diff
@@ -88,15 +102,47 @@ class="tribe-events-calendar-month__day-date-daynum"
...
+				?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: * :no_entry_sign: Line indented incorrectly; expected 6 tabs, found 4
* :no_entry_sign: Closing PHP tag must be on a line by itself

<sub>hash: 49d22fda2fa25bc0da4eb438b91d5ef0</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 30]
```diff
@@ -108,7 +154,18 @@ class="tribe-events-calendar-month__day-cell tribe-events-calendar-month__day-ce
 	>
 		<h3 class="tribe-events-calendar-month__day-date tribe-common-h4">
 			<span class="tribe-common-a11y-visual-hide">
-				<?php echo esc_html( sprintf( _n( '%s event', '%s events', $day['found_events'], 'the-events-calendar' ), number_format_i18n( $day['found_events'] ) ) ); ?>,
+				<?php
+				echo esc_html(
+					sprintf(
+						/* translators: %1$s: number of events, %2$s: Event (singular), %3$s: Events (plural). */
+						_n( '%1$s %2$s', '%1$s %3$s', $day['found_events'], 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :warning: Mismatched placeholders is probably an error

<sub>hash: 9e5eb7cc5b7289663e3d2905fbeda3be</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 31]
```diff
@@ -55,6 +55,17 @@
 
 // Only add id if events exist on the day.
 $mobile_day_id = 'tribe-events-calendar-mobile-day-' . $day['year_number'] . '-' . $day['month_number'] . '-' . $day['day_number'];
+
+$events_label_singular = tribe_get_event_label_singular();
+$events_label_plural   = tribe_get_event_label_plural();
+
+$num_events_label = sprintf(
+	/* translators: %1$s: number of events, %2$s: Event (singular), %3$s: Events (plural). */
+	_n( '%1$s %2$s', '%1$s %3$s', $day['found_events'], 'the-events-calendar' ),
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: :warning: Mismatched placeholders is probably an error

<sub>hash: c90961326cd3ad605e454d1729393a58</sub>

File: src/views/v2/month/calendar-body/day.php [hunk 32]
```diff
@@ -76,7 +87,7 @@
 	>
 		<h3 class="tribe-events-calendar-month__day-date tribe-common-h6 tribe-common-h--alt">
 			<span class="tribe-common-a11y-visual-hide">
-				<?php echo esc_html( sprintf( _n( '%s event', '%s events', $day['found_events'], 'the-events-calendar' ), number_format_i18n( $day['found_events'] ) ) ); ?>,
+				<?php echo esc_html( $num_events_label); ?>,
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day.php]: * :no_entry_sign: Expected 1 spaces before closing bracket; 0 found
* :no_entry_sign: There must be a single space before the last parenthesis

<sub>hash: 20f9efaab55d9998a28b0223a498fdcd</sub>

File: src/Tribe/Views/V2/Messages.php (hunk 1 above)
Comment (lucatume) [reply src/Tribe/Views/V2/Messages.php]: After a hangout about this, I think Paul's solution is a good balance between localization, white-labeling and maintainability concerns.

**Pull request merged**